les/sync.go

```diff
@@ -99,99 +37,13 @@ func (h *clientHandler) synchronise(peer *serverPeer) {
-	)
-	if h.checkpoint != nil && h.checkpoint.SectionIndex >= peer.checkpoint.SectionIndex {
-		local, checkpoint = true, h.checkpoint
-	}
-	// Replace the checkpoint with locally configured one If it's required by
-	// users. Nil checkpoint means synchronization from the scratch.
-	if h.backend.config.SyncFromCheckpoint {
-		local, checkpoint = true, h.backend.config.Checkpoint
-		if h.backend.config.Checkpoint == nil {
-			checkpoint = &params.TrustedCheckpoint{}
-		}
-	}
-	// Determine whether we should run checkpoint syncing or normal light syncing.
-	//
-	// Here has four situations that we will disable the checkpoint syncing:
-	//
-	// 1. The checkpoint is empty
-	// 2. The latest head block of the local chain is above the checkpoint.
-	// 3. The checkpoint is local(replaced with local checkpoint)
-	// 4. For some networks the checkpoint syncing is not activated.
-	mode := checkpointSync
-	switch {
-	case checkpoint.Empty():
-		mode = lightSync
-		log.Debug("Disable checkpoint syncing", "reason", "empty checkpoint")
-	case latest.Number.Uint64() >= (checkpoint.SectionIndex+1)*h.backend.iConfig.ChtSize-1:
-		mode = lightSync
-		log.Debug("Disable checkpoint syncing", "reason", "local chain beyond the checkpoint")
-	case local:
-		mode = legacyCheckpointSync
-		log.Debug("Disable checkpoint syncing", "reason", "checkpoint is hardcoded")
-	case h.backend.oracle == nil || !h.backend.oracle.IsRunning():
-		if h.checkpoint == nil {
-			mode = lightSync // Downgrade to light sync unfortunately.
-		} else {
-			checkpoint = h.checkpoint
-			mode = legacyCheckpointSync
-		}
-		log.Debug("Disable checkpoint syncing", "reason", "checkpoint syncing is not activated")
-	}
-
-	// Notify testing framework if syncing has completed(for testing purpose).
+	// Notify testing framework if syncing has completed (for testing purpose).
 	defer func() {
 		if h.syncEnd != nil {
 			h.syncEnd(h.backend.blockchain.CurrentHeader())
 		}
 	}()
-
 	start := time.Now()
-	if mode == checkpointSync || mode == legacyCheckpointSync {
-		// Validate the advertised checkpoint
-		if mode == checkpointSync {
-			if err := h.validateCheckpoint(peer); err != nil {
-				log.Debug("Failed to validate checkpoint", "reason", err)
-				h.removePeer(peer.id)
-				return
-			}
-			h.backend.blockchain.AddTrustedCheckpoint(checkpoint)
-		}
-		log.Debug("Checkpoint syncing start", "peer", peer.id, "checkpoint", checkpoint.SectionIndex)
-
-		// Fetch the start point block header.
-		//
-		// For the ethash consensus engine, the start header is the block header
-		// of the checkpoint.
-		//
-		// For the clique consensus engine, the start header is the block header
-		// of the latest epoch covered by checkpoint.
-		ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
-		defer cancel()
-		if !checkpoint.Empty() && !h.backend.blockchain.SyncCheckpoint(ctx, checkpoint) {
-			log.Debug("Sync checkpoint failed")
-			h.removePeer(peer.id)
-			return
-		}
-	}
-
```

Comment:
Just to clarity, light client has to sync headers since genesis by running light sync.
But light client is not working anyway, I think it's fine.